api/src/main/java/net/countercraft/movecraft/TrackedLocation.java

```diff
@@ -4,44 +4,81 @@
 import net.countercraft.movecraft.util.MathUtils;
 import org.jetbrains.annotations.NotNull;
 
+import java.lang.ref.WeakReference;
+
 public class TrackedLocation {
-    private MovecraftLocation offSet;
-    private final Craft craft;
+
+    private int dx;
+    private int dy;
+    private int dz;
```

Comment:
Why not use a `MovecraftLocation`?

Reply:
Cause thats is immutable

Reply:
And it adds additional overhead i dont need

Reply:
Changed it to use a movecraftlocation instead of the custom class